### PR TITLE
Set state from URL on page load.

### PR DESCRIPTION
**Pull request by jimmyfnugent**

### CODE REVIEW

File: client/src/thunks.ts [hunk 1]
```diff
@@ -208,3 +217,15 @@ export const thunkChangePageSize: AppThunkActionCreator = (size: PageSize) => {
     dispatch(thunkFetchBits());
   };
 };
+
+export const thunkSetStateFromUrlBar: AppThunkActionCreator = () => {
+  return async (dispatch, getState) => {
+    dispatch(setPageSize(getPageSize(history.location.search)));
```
Comment (hopskipnfall) [inline client/src/thunks.ts:223]: i kind of prefer

```typescript
const urlParams = parseUrlParams(history.location);
if (urlParams.pageSize) {
  dispatch(setPageSize(urlParams.pageSize));
}
```

but this works too

File: client/src/uri_util.ts [hunk 2]
```diff
@@ -13,42 +13,49 @@ import {
...
+export const getStageFilters = (query: string) =>
+  paramStringToFilters(queryStringToObject(query)[QUERY_STAGES], STAGE_MAP);
+
+export const getTagFilters = (query: string) =>
+  paramStringToFilters(queryStringToObject(query)[QUERY_TAGS], LABEL_MAP);
+
+export const getSort = (query: string) =>
+  queryStringToObject(query)[QUERY_SORT];
+
+export const getPageSize = (query: string) =>
```
Comment (hopskipnfall) [inline client/src/uri_util.ts:53]: The method where we parse the URL for a page size doesn't seem like the right place to put the page size default.

**Review by hopskipnfall — APPROVED** [review]

File: client/src/thunks.ts (hunk 1 above)
Comment (jimmyfnugent) [reply client/src/thunks.ts:223]: Well you're right to check that urlParams.pageSize is defined first -- turns out that's actually a bug here that I'll send a PR for momentarily. The one method to parse all URL params is probably a good idea, but going to add a TODO to refactor that.

File: client/src/uri_util.ts (hunk 2 above)
Comment (jimmyfnugent) [reply client/src/uri_util.ts:53]: So this actually isn't the default, but the radix (lol). Commented.

**Pull request merged**